aggregator_core/src/datastore/test_util.rs

```diff
@@ -49,24 +59,73 @@ impl EphemeralDatabase {
     }
 
     async fn start() -> Self {
-        // Start an instance of Postgres running in a container.
-        let db_container = ContainerRequest::from(Postgres::default())
-            .with_cmd(Vec::from([
-                "-c".to_string(),
-                "max_connections=200".to_string(),
-            ]))
-            .start()
-            .await
-            .unwrap();
-        const POSTGRES_DEFAULT_PORT: u16 = 5432;
-        let port_number = db_container
-            .get_host_port_ipv4(POSTGRES_DEFAULT_PORT)
-            .await
-            .unwrap();
-        trace!("Postgres container is up with port {port_number}");
-
+        let (shutdown_tx, shutdown_rx) = oneshot::channel();
+        let (port_tx, port_rx) = oneshot::channel();
+
+        // Hack: run testcontainer logic under its own thread with its own tokio runtime, to avoid
+        // deadlocking the main runtime when waiting for logs to finish in the Drop implementation.
+        let db_thread = std::thread::spawn(move || {
+            tokio::runtime::Builder::new_current_thread()
```

Comment:
This approach to resolving the deadlock LGTM. I suspect another way to solve it would be with `[tokio::test(flavor = "multi_threaded")]` everywhere, but that would be very intrusive.

Reply:
I think you're right, if all tests run multithreaded, then maybe we can use `block_in_place()`. But yeah having to decorate every test would be irritating--we would probably be better off writing our own macro wrapper in that case. (I wonder why tokio tests default to the current thread runtime anyways.)